main/main.cpp

```diff
@@ -3254,9 +3267,15 @@ bool Main::iteration() {
 					"Command line option --build-solutions was passed, but the build callback failed. Aborting.");
 		}
 	}
+
+	if (wait_for_import) {
+		if (EditorFileSystem::get_singleton() && !EditorFileSystem::get_singleton()->is_first_scan() && !EditorFileSystem::get_singleton()->is_importing()) {
+			should_quit = true;
```

Comment:
Does this PR still work if this line is changed and the other changes involving `should_quit` and `auto_quit` are dropped?
```diff
--- main.cpp	2023-02-22 09:59:39.998880978 -0500
+++ main.cpp2	2023-02-22 10:00:39.927385543 -0500
@@ -1197,7 +1197,6 @@
 			wait_for_import = true;
 			cmdline_tool = true;
 			editor = true;
-			auto_quit = true;
 #endif
 		} else if (I->get() == "--quit") { // Auto quit at the end of the first main loop iteration
 			auto_quit = true;
@@ -3251,7 +3250,6 @@
 
 	OS::get_singleton()->add_frame_delay(DisplayServer::get_singleton()->window_can_draw());
 
-	bool should_quit = (exit || auto_quit) && !wait_for_import;
 #ifdef TOOLS_ENABLED
 	if (auto_build_solutions) {
 		auto_build_solutions = false;
@@ -3270,12 +3268,12 @@
 
 	if (wait_for_import) {
 		if (EditorFileSystem::get_singleton() && !EditorFileSystem::get_singleton()->is_first_scan() && !EditorFileSystem::get_singleton()->is_importing()) {
-			should_quit = true;
+			exit = true;
 		}
 	}
 #endif
 
-	return should_quit;
+	return exit || auto_quit;
 }
 
 void Main::force_redraw() {
```

As-is, this conflicts with #73617 .